src/client/refresh/soft/sw_main.c

```diff
@@ -1680,17 +1632,26 @@ CreateSDLWindow(int flags, int w, int h)
 		renderer = SDL_CreateRenderer(window, -1, SDL_RENDERER_ACCELERATED);
 	}
 
-	surface = SDL_CreateRGBSurface(0, w, h, bpp, Rmask, Gmask, Bmask, Amask);
+	/* Select the color for drawing. It is set to black here. */
+	SDL_SetRenderDrawColor(renderer, 0, 0, 0, 0);
+
+	/* Clear the entire screen to our selected color. */
+	SDL_RenderClear(renderer);
+
+	/* Up until now everything was drawn behind the scenes.
+	   This will show the new, black contents of the window. */
+	SDL_RenderPresent(renderer);
 
 	texture = SDL_CreateTexture(renderer,
-				    SDL_PIXELFORMAT_ARGB8888,
+				    SDL_PIXELFORMAT_BGRA32,
```

Comment:
SDL_PIXELFORMAT_BGRA32 seems to be a relatively new addition to SDL2.

I'd suggest using something like:
```c
int pixelformat;
#if SDL_BYTEORDER == SDL_BIG_ENDIAN
    pixelformat = SDL_PIXELFORMAT_BGRA8888;
#else
    pixelformat = SDL_PIXELFORMAT_ARGB8888;
#endif
```
and then pass the `pixelformat` variable to `SDL_CreateTexture()`

It's a bit uglier, but at least it works with older SDL2 versions, including the one used by Ubuntu 16.04, Linux Mint 18.x and other slightly old Linux distros.